packages/@aws-cdk/aws-stepfunctions-tasks/lib/codebuild/start-build.ts

```diff
@@ -12,12 +125,109 @@ export interface CodeBuildStartBuildProps extends sfn.TaskStateBaseProps {
+   */
+  readonly cacheOverride?: codebuild.Cache;
+  /**
+   * The name of a certificate for this build.
+   *
+   * @default - the certificate specified in the build project.
+   */
+  readonly certificateOverride?: string;
+  /**
+   * Specifies if session debugging is enabled for this build.
+   *
+   * @default - the debug session flag specified in the build project.
+   */
+  readonly debugSessionEnabled?: boolean;
+  /**
+   * The AWS Key Management Service (AWS KMS) customer master key (CMK).
+   *
+   * @default - the encryption key specified in the build project.
+   */
+  readonly encryptionKeyOverride?: kms.IKey;
+  /**
+   * BuildEnvironment includes build image, privilege, environment variable that overrides, for this build only.
+   *
+   * @default - the build environment specified in the build project.
+   */
+  readonly environmentOverride?: codebuild.BuildEnvironment;
+  /**
+   * A container type for this build.
+   *
+   * @default - the environment type specified in the build project.
+   */
+  readonly environmentTypeOverride?: ImageType;
   /**
    * A set of environment variables to be used for this build only.
    *
+   * @deprecated - use {@link environmentOverride} instead
    * @default - the latest environment variables already defined in the build project.
    */
   readonly environmentVariablesOverride?: { [name: string]: codebuild.BuildEnvironmentVariable };
+  /**
+   * A unique, case sensitive identifier you provide to ensure the idempotency of the StartBuild request.
+   *
+   * @default - no idempotency token.
+   */
+  readonly idempotencyToken?: string;
+  /**
+   * Log settings for this build.
+   *
+   * @default - the log config specified in the build project.
+   */
+  readonly logsConfigOverride?: BuildLogConfig;
+  /**
+   * The number of minutes a build is allowed to be queued before it times out.
+   *
+   * @default - the queued timeout specified in the build project.
+   */
+  readonly queuedTimeoutOverride?: cdk.Duration;
+  /**
+   * An array of ProjectArtifacts objects.
+   *
+   * @default - the secondary artifacts specified in the build project.
+   */
+  readonly secondaryArtifactsOverride?: [codebuild.IArtifacts];
+  /**
+   * An array of ProjectSource objects.
+   *
+   * @default - the second sources specified in the build project.
+   */
+  readonly secondarySourcesOverride?: [codebuild.ISource];
+  /**
+   * The name of a service role for this build.
+   *
+   * @default - the name of service role specified in the build project.
+   */
+  readonly serviceRoleOverride?: iam.IRole;
+  /**
+   * The source type, source version defined in the build project.
+   *
+   * @default - the source specified in the build project.
+   */
+  readonly sourceOverride?: codebuild.ISource;
+  /**
+   * The number of build timeout minutes, from 5 to 480 (8 hours).
+   *
+   * @default - the build timeout specified in the build project.
+   */
+  readonly timeoutOverride?: cdk.Duration;
```

Comment:
All of these properties should be modeled in a nested interface, not on this level. So the usage would look like:

```ts
new sfn_tasks.CodeBuildStartBuild(this, 'StartBuild', {
  overrides: {
    timeout: cdk.Duration.minutes(5),
  },
});
```